### PR TITLE
Fix UpdateResultsAPI duplicate record issue

### PR DESCRIPTION
**Pull request by khansaad**
This PR fixes the duplicate record mismatch issue after hitting the updateResults API. We were seeing a difference in the number of records getting saved in the DB.

### CODE REVIEW

**Comment by dinogun** [comment]
@khansaad as mentioned we need to add the appropriate negative tests to handle this scenario. @chandrams please review

**Review by msvinaykumar — APPROVED** [review]

File: tests/scripts/remote_monitoring_tests/rest_apis/test_update_results.py [hunk 1]
```diff
@@ -719,3 +719,52 @@ def test_update_results_with_valid_and_invalid_interval_duration(test_name, inte
...
+    response = update_results(result_json_file)
+
+    # Get the experiment name
+    json_data = json.load(open(input_json_file))
+    experiment_name = json_data[0]['experiment_name']
+
+    data = response.json()
+    assert response.status_code == ERROR_STATUS_CODE
+    assert data['status'] == ERROR_STATUS
+    assert data['message'] == UPDATE_RESULTS_FAILED_RECORDS_MSG
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_update_results.py:754]: Also validate the failed records data error messages - "An entry for this record already exists!"

**Review by chandrams — CHANGES_REQUESTED** [review]

File: tests/scripts/remote_monitoring_tests/rest_apis/test_update_results.py [hunk 2]
```diff
@@ -719,3 +719,60 @@ def test_update_results_with_valid_and_invalid_interval_duration(test_name, inte
...
+    # Get the experiment name
+    json_data = json.load(open(input_json_file))
+    experiment_name = json_data[0]['experiment_name']
+
+    data = response.json()
+    # Extract the inner 'message' from each 'errors' object and then verify it
+    error_messages = [error['message'] for item in data['data'] for error in item['errors']]
+    error_messages_code = [error['httpcode'] for item in data['data'] for error in item['errors']]
+
+    assert DUPLICATE_RECORDS_MSG in error_messages
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_update_results.py:756]: We need to validate each error message or http code and use equals here

File: tests/scripts/remote_monitoring_tests/rest_apis/test_update_results.py [hunk 3]
```diff
@@ -719,3 +719,60 @@ def test_update_results_with_valid_and_invalid_interval_duration(test_name, inte
...
+    assert ERROR_409_STATUS_CODE in error_messages_code
+
+    assert response.status_code == ERROR_STATUS_CODE
+    assert data['status'] == ERROR_STATUS
+    assert data['message'] == UPDATE_RESULTS_FAILED_RECORDS_MSG
+
+    response = list_experiments("true", None, "false", experiment_name)
```
Comment (chandrams) [inline tests/scripts/remote_monitoring_tests/rest_apis/test_update_results.py:763]: Assign the values to variables and pass those variables for better readability. Also update any other references to list_experiments in other tests.

**Review by chandrams — APPROVED** [review]
LGTM

**Pull request merged**